tox.ini

```diff
@@ -4,20 +4,42 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py35,py36,py37,py38,pypy3,flake8
+envlist = flake8,packaging,docs,py3.5,py3.6,py3.7,py3.8,pypy3
 
 [testenv]
 basepython =
-    py35: python3.5
-    py36: python3.6
-    py37: python3.7
-    py38: python3.8
+    py3.5: python3.5
+    py3.6: python3.6
+    py3.7: python3.7
+    py3.8: python3.8
     pypy3: pypy3
-
-commands = {envpython} setup.py test
+deps =
+    pytest
+    pytest-tldr
+whitelist_externals =
+    make
+commands =
+    make
+    pytest -vv
 
 [testenv:flake8]
 basepython = python3.8
 deps =
     flake8
 commands = flake8 {posargs}
+
+[testenv:packaging]
+basepython = python3.8
+deps =
+    wheel
+    twine
+commands =
+    {envpython} setup.py sdist bdist_wheel
+    {envpython} -m twine check dist/*
```

Comment:
In my experience, using `{envpython}` to call Python seems to do more harm than good. `{envpython}` is a textual substitution that is performed by tox when it reads the config, which causes problems with argument splitting when the path to the project contains spaces. ([I have been bitten by this in the past.](https://github.com/dgelessus/python-rsrcfork/commit/007d15eb3d369ad8d0006e853e75b567e347998b))

As far as I can tell, it's much better to use plain `python` instead (and that's what every other project I've looked at does, [including tox itself](https://github.com/tox-dev/tox/blob/9d02cd4b7df111a7a69366ea5a498da3025ad234/tox.ini#L78)). All tox commands are run in a virtual environment, so `python` is guaranteed to refer to the correct Python executable, and there is no chance of any argument splitting problems, because `python` is a simple command and not a variable substitution.